rmw_connextdds_common/src/common/rmw_info.cpp

```diff
@@ -538,19 +538,6 @@ rmw_api_connextdds_get_publishers_info_by_topic(
   RMW_CHECK_ARGUMENT_FOR_NULL(topic_name, RMW_RET_INVALID_ARGUMENT);
   RMW_CHECK_ARGUMENT_FOR_NULL(publishers_info, RMW_RET_INVALID_ARGUMENT);
 
-  int validation_result = RMW_TOPIC_VALID;
-  rmw_ret_t ret =
-    rmw_validate_full_topic_name(topic_name, &validation_result, nullptr);
-  if (RMW_RET_OK != ret) {
-    return ret;
-  }
-  if (RMW_TOPIC_VALID != validation_result) {
-    const char * reason =
-      rmw_full_topic_name_validation_result_string(validation_result);
-    RMW_CONNEXT_LOG_ERROR_A_SET("invalid topic name: %s", reason)
-    return RMW_RET_INVALID_ARGUMENT;
-  }
```

Comment:
See [this thread](https://github.com/ros2/rmw_connextdds/pull/22#discussion_r610673301) about these removals. I'm considering restoring these checks, only for `!no_mangle`, but a unit test fails (which is actually the reason why I had removed them in the first place, because that test was failing, and I assumed the test to be correct since I noticed that other RMW implementation don't perform the same validation).